### PR TITLE
feat(web): API Authentication Integration (Login & Register)

### LINKED ISSUE #24: [2JH] API Integration (Login & Register)
## Describe the issue
The users should be able to create an account or login if they already have one

## Resource request
Get access to the API docs

## Definition of done
- [x] Create an account if not existed
- [x] Login if there is an account
- [x] Fetch and manage all the errors (invalid email, incorrect password, account already created...)

## testing plan
Demo with most of the cases

## JEH
2

### PR DESCRIPTION
**Pull request by hamid-yg**
# Description

Please provide a detailed description of what was done in this PR.
Precise the issue that you are resolving.

# Fixes

Specify which issue this PR targets, like so:

- fixes #24 

# Changes include

- [x] Bugfix (non-breaking change that solves an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Breaking changes

Please complete this section if any breaking changes have been made, otherwise delete it.

# Checklist

- [x] I have assigned this PR to myself
- [x] I have added @vrn-sh/members to the reviewers, or specific members of the team
- [x] I have added the needed labels
- [x] I have tested this code
- [x] I have added / updated tests (unit / functionals / end-to-end / ...)
- [x] I have updated the README and other relevant documents (guides...)

# Additional comments

Please post additional comments in this section if you have them, otherwise delete it.

### CODE REVIEW

File: api/core/settings.py [hunk 1]
```diff
@@ -104,8 +105,12 @@
     'django.contrib.messages.middleware.MessageMiddleware',
     'django.middleware.clickjacking.XFrameOptionsMiddleware',
     'api.management.middlewares.set_secure_headers',
+    'corsheaders.middleware.CorsMiddleware',
 ]
 
+ALLOWED_HOSTS=['*']
```
Comment (djnnvx) [inline api/core/settings.py]: This is very dangerous. Using `'*'` as allowed hosts is only OK for Local development, I can fix this in this branch if that is ok for you @blacky-yg ?

File: web/.env.example
Comment (djnnvx) [inline web/.env.example:1]: It could be good to rename this to `env-dist`, like the one used for the API.

**Review by djnnvx — CHANGES_REQUESTED** [review]
Having CORS configuration set with any Host is a security issue. I can fix this if you don't have the time, but we should not merge this until it is fixed.

**Comment by djnnvx** [comment]
ok to review @claraxuxu

**Review by djnnvx — APPROVED** [review]

**Review by djnnvx — APPROVED** [review]

**Review by claraxuxu — APPROVED** [review]

**Comment by djnnvx** [comment]
good work @blacky-yg

**Pull request merged**